src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -458,39 +462,55 @@ build_mhd_response(const struct sol_http_response *response, time_t last_modifie
 static void
 progressive_response_del_cb(void *data)
 {
+    struct sol_blob *blob;
+    uint16_t i;
     struct sol_http_progressive_response *progressive = data;
 
-    if (progressive->on_del)
-        progressive->on_del((void *)progressive->cb_data, progressive);
+    SOL_PTR_VECTOR_FOREACH_IDX (&progressive->pending_blobs, blob, i) {
+        if (progressive->on_feed)
+            progressive->on_feed((void *)progressive->cb_data, progressive, blob, -ECANCELED);
```

Comment:
We're fine here. the `delete_me` variable will be true and `sol_http_progressive_response_feed` will fail